### PR TITLE
Refactor container brokers from worker/provisioner into container/broker

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

To facilitate upcoming work to make a worker for lxd profile consistency on machines/containers, move the container brokers out of the provisioner package and into their own.

## QA steps

Bootstrap and deploy charms to both lxd machines and containers.  Preferability with lxd profiles.  Upgrade the units as well.

## Documentation changes

n/a

## Bug reference

n/a

### CODE REVIEW

File: container/broker/broker.go [hunk 1]
```diff
@@ -30,16 +31,6 @@ type APICalls interface {
 	HostChangesForContainer(containerTag names.MachineTag) ([]network.DeviceToBridge, int, error)
 }
 
-type hostArchToolsFinder struct {
```
Comment (hmlanigan) [inline container/broker/broker.go]: unused code

File: container/broker/kvm-broker_test.go [hunk 2]
```diff
@@ -328,155 +322,3 @@ func (s *kvmBrokerSuite) TestStartInstanceWithContainerInheritProperties(c *gc.C
 		},
 	}, c)
 }
-
```
Comment (hmlanigan) [inline container/broker/kvm-broker_test.go]: The kvmProvisionerSuite covers tests for the container provisioner using a kvm container broker.  Moved to a new file in worker/provisioner.

File: worker/provisioner/containerprovisioner_test.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	}
+}
+
+func (s *kvmProvisionerSuite) TearDownTest(c *gc.C) {
+	close(s.events)
+	s.TestSuite.TearDownTest(c)
+	s.CommonProvisionerSuite.TearDownTest(c)
+}
+
+func (s *kvmProvisionerSuite) newKvmProvisioner(c *gc.C) provisioner.Provisioner {
```
Comment (hmlanigan) [inline worker/provisioner/containerprovisioner_test.go]: Cover container provisioner unit tests.  Use a mock broker to as the KVM broker to start it.

File: worker/provisioner/export_test.go [hunk 4]
```diff
@@ -38,10 +38,8 @@ func GetRetryWatcher(p Provisioner) (watcher.NotifyWatcher, error) {
 var (
 	GetContainerInitialiser = &getContainerInitialiser
 	GetToolsFinder          = &getToolsFinder
-	ResolvConfFiles         = &resolvConfFiles
```
Comment (hmlanigan) [inline worker/provisioner/export_test.go]: ResolvConfFiles and CombinedCloudInitData only used by the container broker tests, moved.

**Comment by hmlanigan** [comment]
[refactor container brokers out of the provisioner worker.](https://trello.com/c/DKzTfEMt/1116-refactor-container-brokers-out-of-the-provisioner-worker)

File: container/broker/kvm-broker.go [hunk 5]
```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/juju/juju/network"
 )
 
-var kvmLogger = loggo.GetLogger("juju.provisioner.kvm")
+var kvmLogger = loggo.GetLogger("juju.container.broker.kvm")
```
Comment (SimonRichardson) [inline container/broker/kvm-broker.go]: If people look out/script against debug-log output should we make them aware of this?

File: container/broker/lxd-broker.go [hunk 6]
```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/juju/juju/provider/common"
 )
 
-var lxdLogger = loggo.GetLogger("juju.provisioner.lxd")
+var lxdLogger = loggo.GetLogger("juju.container.broker.lxd")
```
Comment (SimonRichardson) [inline container/broker/lxd-broker.go]: see output comment

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

> Failure: CAASProvisionerSuite.TestUpdateApplicationsUnitsWithStorage

**Comment by SimonRichardson** [comment]
$$merge$$

> Failure: apiclientSuite.TestOpenDialTimeoutAffectsDial

**Pull request merged**